cardano-cli/test/cardano-cli-test/Test/Cli/CreateTestnetData.hs

```diff
@@ -19,11 +34,60 @@ hprop_create_testnet_data_minimal =
 
     let outputDir = tempDir </> "out"
 
+    -- We test that the command doesn't crash, because otherwise
+    -- execCardanoCLI would fail.
     H.noteM_ $ execCardanoCLI
       ["conway",  "genesis", "create-testnet-data"
       , "--testnet-magic", "42"
       , "--out-dir", outputDir
       ]
+    success
 
-    -- We test that the command doesn't crash, because otherwise
-    -- execCardanoCLI would fail.
+hprop_create_testnet_data_create_nonegative_supply :: Property
+hprop_create_testnet_data_create_nonegative_supply = do
+  -- FIXME rewrite this as a property test
+  let supplyValues =
+        [ -- (total supply, delegated supply)
+          (2_000_000_000, 1_000_000_000)
+        , (1_000_000_000, 1_000_000_000)
+        , (1_000_000_000, 2_000_000_000)
```

Comment:
@carbolymer> when _delegated supply_ is bigger than _total supply_, then `create-testnet-data` should crash. I can do that in a follow-up PR. What do you think?

Reply:
I've added error throwing when created funds are negative.